app/controllers/ems_common.rb

```diff
@@ -165,6 +165,39 @@ def edit
                     :url  => "/#{controller_name}/#{@ems.id}/edit")
   end
 
+  def properties
+    # Get the query parameters:
+    id = params[:id]
+    type = params[:type]
+
+    # Assert the permissions:
+    permission_suffix = if id.blank? then :new else :edit end
```

Comment:
nitpick: can be written as:
```ruby
permission_suffix = id.blank? ? :new : :edit
```
and just to be consistent with permissions_prefix, maybe worth extracting into a method.

Reply:
Extracting it to a method won't make it as nice as `permissions_prefix`, because it would need to receive the `id` as parameter. I will leave it as it is, unless you consider it important.